frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -151,62 +208,110 @@ export class MarketplaceTileViewPage extends React.Component {
     return items;
   }
 
-  getFilterCountsFromGroups(items, filters, itemsFilteredByGroup) {
-    // Input items, filters, and mapping from filter property to filtered items
-    // Returns object mapping filter property to item counts
-    return _.reduce(filters, (counts, group, key) => {
-      if (key === 'keyword') {
-        return counts;
-      }
+  static getCurrentCategories(activeTabId, categories) {
+    const selectedCategory = CategoryFilterUtils.findActiveCategory(activeTabId, categories);
+    return (selectedCategory && selectedCategory.subcategories) || categories;
+  }
 
-      // Apply filters for all groups except for current group key
-      const itemsByGroup = _.filter(itemsFilteredByGroup, (groupItems, groupName) => groupName !== key);
-      const count = _.countBy(_.intersection(...itemsByGroup, items), key);
+  clearFilters() {
+    const params = new URLSearchParams(window.location.search);
 
-      counts[key] = _.reduce(group, (groupCounts, value, elem) => {
-        groupCounts[elem] = count[elem] || 0;
-        return groupCounts;
-      }, {});
+    params.delete('keyword');
 
-      return counts;
-    }, {});
+    _.forEach(filterGroups, filterGroup => {
+      params.delete(filterGroup);
+    });
+
+    const url = new URL(window.location);
+    const searchParams = `?${params.toString()}${url.hash}`;
+
+    history.replace(`${url.pathname}${searchParams}`);
+    this.setState(this.getUpdatedStateFromURL(searchParams));
```

Comment:
After clearing filters the keyword textbox should get focus.

Reply:
Thanks. Fixed.